Acme/Certificate/Storage/CertificateStorage.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+/*
+ * This file is part of the ACME PHP library.
+ *
+ * (c) Titouan Galopin <galopintitouan@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace AcmePhp\Bundle\Acme\Certificate\Storage;
+
+use AcmePhp\Bundle\Exception\FormatNotFoundException;
+use Symfony\Component\Filesystem\Filesystem;
+
+/**
+ * Certificates storage.
+ *
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ */
+class CertificateStorage
+{
+    /** @var Filesystem */
+    private $filesystem;
+
+    /** @var string */
+    private $storagePath;
+
+    /** @var string */
+    private $backupStoragePath;
+
+    /**
+     * @param Filesystem $filesystem
+     * @param string     $storagePath
+     * @param string     $backupStoragePath
+     */
+    public function __construct(Filesystem $filesystem, $storagePath, $backupStoragePath)
+    {
+        $this->filesystem = $filesystem;
+        $this->storagePath = $storagePath;
+        $this->backupStoragePath = $backupStoragePath;
+    }
+
+    /**
+     * Backup the certificates files into the configured `backupStoragePath`.
+     */
+    public function backup()
+    {
+        $this->filesystem->mirror(
+            $this->storagePath,
+            $this->backupStoragePath.DIRECTORY_SEPARATOR.date('Ymd-His')
+        );
+    }
+
+    /**
+     * Returns whether or not a formatted certificate exists.
+     *
+     * @param string $formatName
+     *
+     * @return bool
+     */
+    public function hasFormat($formatName)
```

Comment:
There is so much certificates in this project :/

Nb: I don't like `format` too